src/service/resources-impl.js

```diff
@@ -345,6 +345,8 @@ export class Resources {
     if (!element.id) {
       element.id = 'AMP_' + resource.getId();
     }
+    element.ampResourceId = resource.getId();
```

Comment:
I meant more like this in `custom-element.js`:

```
BaseCustomElement {
...
  getResourceId() {
    return this.getResources().getResourceForElement(this).getId();
  }
}
```

Although I do kind of like the `ampResourceId`. If you do that - we need to also update externs I think?